tests/test_agents.py

```diff
@@ -161,10 +160,11 @@ async def crashing_aadd(*args, **kwargs) -> str | None:
             ) as mock_aadd,
         ):
             index = await get_directory_index(settings=agent_test_settings)
+
+    assert len(await index.index_files) == num_source_files
     assert (
-        mock_aadd.await_count <= crash_threshold
+        mock_aadd.await_count != num_source_files
```

Comment:
Can we do `< num_source_files` here? `!=` is less specific.

Also, can you beef up the comment `"Should have been able to resume build"` to better reflect the assertion?